File: 00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let reversedString = "";

  for (let i = str.length - 1; i >= 0; i--) {
    reversedString += str[i];
  }

  return reversedString;
}

if (require.main === module) {
  // add your own tests in here
  console.log("hello world: ", reverseString("hello world"))
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
  //create empty string
  //loop for length of string
     //add character from str to empty string
  //return string

File: 00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  reversed_str = ""
  str.each_char { |c| reversed_str = c + reversed_str }
  reversed_str
  # type your code in here
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts  "hi ","=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "catbaby ","=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution